### PR TITLE
[I18N] edit locale modal

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

- Provides the logic for editing a locale
- Add the tests to cover the management rules
- TODO (another PR?): make the UI

### CODE REVIEW

File: packages/strapi-plugin-i18n/admin/src/components/LocaleList/index.js [hunk 1]
```diff
@@ -7,11 +7,25 @@ import PropTypes from 'prop-types';
 import { useLocales } from '../../hooks';
 import LocaleRow from '../LocaleRow';
 import { getTrad } from '../../utils';
+import ModalEdit from '../ModalEdit';
+import ModalDelete from '../ModalDelete';
 
-const LocaleList = ({ onAddLocale, onDeleteLocale, onEditLocale }) => {
-  const { locales, isLoading } = useLocales();
+const LocaleList = ({ canUpdateLocale, canDeleteLocale, canCreateLocale }) => {
+  const [localeToDelete, setLocaleToDelete] = useState(undefined);
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/LocaleList/index.js:14]: Why do you need to explicitly set the default value to `undefined`. Isn't that already `undefined` by default?

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js [hunk 2]
```diff
@@ -2,51 +2,109 @@ import React from 'react';
...
 import { getTrad } from '../../utils';
 
-const ModalEdit = ({ localeToEdit, onClose }) => {
+const ModalEdit = ({ localeToEdit, onClose, locales }) => {
   const { isEditing, editLocale } = useEditLocale();
   const { formatMessage } = useIntl();
   const isOpened = Boolean(localeToEdit);
 
-  const handleEdit = () => editLocale(localeToEdit).then(onClose);
+  const handleSubmit = async ({ displayName }) => {
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:17]: do you still need the async keyword for this function?

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/tests/SettingsPage.test.js [hunk 3]
```diff
@@ -163,6 +169,159 @@ describe('i18n settings page', () => {
...
+          type: 'success',
+          message: { id: 'Settings.locales.modal.edit.success' },
+        })
+      );
+
+      expect(request).toBeCalledWith('/i18n/locales/2', {
+        method: 'PUT',
+        body: { name: 'en-US' },
+      });
+    });
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/tests/SettingsPage.test.js:345]: 👍🏽

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-plugin-i18n/admin/src/components/LocaleList/index.js (hunk 1 above)
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/LocaleList/index.js:14]: Yep, you're right, it's not necessary 😊

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js (hunk 2 above)
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:17]: Nice catch 😊 I don't need this one 👍🏻 , thank you 🙏🏻

File: packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/tests/SettingsPage.test.js [hunk 4]
```diff
@@ -147,11 +147,32 @@ describe('i18n settings page', () => {
...
+      );
+
+      const row = await waitFor(() => screen.getByText('English').closest('tr'));
+      const rowUtils = within(row);
+
+      fireEvent.click(rowUtils.getByLabelText('Settings.list.actions.edit'));
+      fireEvent.change(screen.getByLabelText('Settings.locales.modal.edit.locales.displayName'), {
+        target: {
+          value:
+            'a very very very very long string that has more than fifty characters in order to show a warning',
```
Comment (soupette) [inline packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/tests/SettingsPage.test.js:166]: 😆

**Review by soupette — APPROVED** [review]
LGTM!

**Pull request merged**